src/pages/HabitTracker/Habits.tsx

```diff
@@ -0,0 +1,224 @@
+    setOriginalHabitName("");
+    setOriginalHabitCount("");
+  };
+
+  // Handler for saving a new or edited habit
+  const handleSaveHabit = () => {
+    if (newHabitName && newHabitCount) {
+      const updatedHabits = [...habits];
+      // Update the habit if editing
+      if (editIndex !== null) {
+        updatedHabits[editIndex] = {
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: editIndex + 1,
+        };
+      } else {
+        // Add the new habit
+        updatedHabits.push({
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: habits.length + 1,
+        });
+      }
+      setHabits(updatedHabits);
+      localStorage.setItem("habits", JSON.stringify(updatedHabits));
+      setShowModal(false);
+      resetForm();
+    }
+  };
+
+  // Handler for closing the modal without saving
+  const handleCloseModal = () => {
+    setShowModal(false);
+    resetForm();
+  };
+
+  // Handler for deleting a habit
+  const handleDeleteHabit = (index: number) => {
+    // Filter out the deleted habit and reindex the remaining habits
+    const updatedHabits = habits
+      .filter((habit) => habit.index !== index)
+      .map((habit, idx) => ({ ...habit, index: idx + 1 }));
+    setHabits(updatedHabits);
+    localStorage.setItem("habits", JSON.stringify(updatedHabits));
+  };
+
+  // Handler for editing a habit
+  const handleEditHabit = (index: number) => {
+    const habitToEdit = habits.find((habit) => habit.index === index);
+    if (habitToEdit) {
+      setNewHabitName(habitToEdit.name);
+      setNewHabitCount(habitToEdit.count.toString());
+      setOriginalHabitName(habitToEdit.name);
+      setOriginalHabitCount(habitToEdit.count.toString());
+      setEditIndex(index - 1);
+      setShowModal(true);
+    }
+  };
+
+  // Handler for habit name input change with validation
+  const handleHabitNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    if (/^[a-zA-Z\s]*$/.test(value)) {
+      setNewHabitName(value);
+    }
+  };
+
+  // Handler for habit count input change with validation
+  const handleHabitCountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    if (/^\d*$/.test(value)) {
+      setNewHabitCount(value);
+    }
+  };
+
+  // Check if the Save button should be enabled
+  const isSaveDisabled = () =>
+    !newHabitName ||
+    !newHabitCount ||
+    (editIndex !== null &&
+      newHabitName === originalHabitName &&
+      newHabitCount === originalHabitCount);
+
+  return (
+    <PageBackground>
+      <Sidebar links={links} />
+      <HabitListContainer>
+        <HabitList>
+          <AddHabitTile onAddClick={handleAddHabitClick} />
+          {habits.map((habit) => (
+            <HabitTile
+              key={habit.index}
+              habitName={habit.name}
+              habitIndex={habit.index}
+              habitCount={habit.count}
```

Comment:
instead of passing each parameter of `habit` like this, why not pass the complete `habit` object?